newIDE/app/src/Profile/Subscription/SubscriptionDetails.js

```diff
@@ -110,28 +116,106 @@ const SubscriptionDetails = ({
   );
   const { showAlert } = useAlertDialog();
   const gdevelopTheme = React.useContext(GDevelopThemeContext);
+  const [
+    userSubscriptionPlan,
+    setUserSubscriptionPlan,
+  ] = React.useState<?SubscriptionPlan>(null);
+  const [error, setError] = React.useState<?React.Node>(null);
+  const [isLoadingUserPrice, setIsLoadingUserPrice] = React.useState<boolean>(
+    false
+  );
+  const [
+    userSubscriptionPlanPrice,
+    setUserSubscriptionPlanPrice,
+  ] = React.useState<?SubscriptionPlanPrice>(null);
 
-  const userPlan = React.useMemo(
+  React.useEffect(
     () => {
-      if (!subscription) return null;
-      const possiblePlans: Array<PlanDetails> = getSubscriptionPlans().concat(
-        getFormerSubscriptionPlans()
-      );
-      return possiblePlans.find(plan => subscription.planId === plan.planId);
+      (async () => {
+        setError(null);
+        setIsLoadingUserPrice(true);
+        try {
+          if (!subscription) {
+            setUserSubscriptionPlan(null);
+            setUserSubscriptionPlanPrice(null);
+            return;
+          }
+
+          const { planId, pricingSystemId } = subscription;
+          if (!planId || !pricingSystemId) {
+            setUserSubscriptionPlan(null);
+            setUserSubscriptionPlanPrice(null);
+            return;
+          }
+
+          const matchingSubscriptionPlanWithPrices = subscriptionPlansWithPrices.find(
+            plan => subscription.planId === plan.id
+          );
+          if (!matchingSubscriptionPlanWithPrices) {
+            setError(
+              <Trans>
+                Couldn't find a subscription matching your account. Please get
+                in touch with us to fix this issue.
+              </Trans>
+            );
+            setUserSubscriptionPlan(null);
+            setUserSubscriptionPlanPrice(null);
+            return;
+          }
+
+          const {
+            prices,
+            ...subscriptionPlan
+          } = matchingSubscriptionPlanWithPrices;
+
+          if (!canPriceBeFoundInGDevelopPrices(pricingSystemId)) {
+            setUserSubscriptionPlan(subscriptionPlan);
+            setUserSubscriptionPlanPrice(null);
+            return;
+          }
+
+          let pricingSystem = prices.find(
```

Comment:
I find the name of this variable confusing, is it a planPrice or a pricingSystem? (I'd go for planPrice)